retz-client/src/main/java/io/github/retz/web/Client.java

```diff
@@ -16,61 +16,38 @@
  */
 package io.github.retz.web;
 
-import com.fasterxml.jackson.databind.ObjectMapper;
-import com.fasterxml.jackson.datatype.jdk8.Jdk8Module;
-import io.github.retz.auth.AuthHeader;
-import io.github.retz.auth.Authenticator;
-import io.github.retz.cli.TimestampHelper;
-import io.github.retz.protocol.*;
-import io.github.retz.protocol.data.Application;
-import io.github.retz.protocol.data.Job;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import javax.xml.bind.DatatypeConverter;
 import java.io.IOException;
-import java.net.HttpURLConnection;
-import java.net.MalformedURLException;
 import java.net.URI;
-import java.net.URL;
-import java.nio.charset.StandardCharsets;
 import java.security.KeyManagementException;
-import java.security.MessageDigest;
 import java.security.NoSuchAlgorithmException;
-import java.util.Objects;
 import java.util.ResourceBundle;
 
-import static java.nio.charset.StandardCharsets.UTF_8;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.github.retz.auth.Authenticator;
+import io.github.retz.protocol.GetJobResponse;
+import io.github.retz.protocol.Response;
+import io.github.retz.protocol.ScheduleResponse;
+import io.github.retz.protocol.data.Application;
+import io.github.retz.protocol.data.Job;
+import io.github.retz.web.feign.Server;
 
 public class Client implements AutoCloseable {
+
     public static final String VERSION_STRING;
+
     static final Logger LOG = LoggerFactory.getLogger(Client.class);
 
     static {
         ResourceBundle labels = ResourceBundle.getBundle("retz-client");
         VERSION_STRING = labels.getString("version");
     }
 
-    private final ObjectMapper mapper;
-    private final String scheme;
-    private final String host;
-    private final int port;
-    private final Authenticator authenticator;
-    private MessageDigest DIGEST = null; //TODO this could be final, though...
+    private final Server server;
 
     protected Client(URI uri, Authenticator authenticator) {
-        this.scheme = uri.getScheme();
-        this.host = uri.getHost();
-        this.port = uri.getPort();
-        this.mapper = new ObjectMapper();
-        this.mapper.registerModule(new Jdk8Module());
-        this.authenticator = authenticator;
-
-        try {
-            DIGEST = MessageDigest.getInstance("MD5");
-        } catch (NoSuchAlgorithmException e) {
-            throw new AssertionError("No such algorithm callled MD5.");
-        }
+        this.server = Server.connect(uri, authenticator);
```

Comment:
The actual connection will be done later but I'd modify to `connect` later because I'm not sure the `server` object is thread-safe or not.